core/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -308,6 +309,39 @@ private static boolean matchAllAliases(final String[] aliases) {
     }
 
     /**
+     * Allows to return the concrete index meta data of the index specified or of the
+     * index that is hidden behind the alias specified
+     *
+     * @param aliasOrIndexName An alias or an index name
+     * @return The index metadata of the concrete index
+     * @throws IllegalArgumentException if the parameter points to an alias with more than one index
+     * @throws IndexNotFoundException if the parameter points neither to an index nor to an alias
+     */
+    public IndexMetaData findConcreteIndexMetaData(String aliasOrIndexName) {
+        assert aliasOrIndexName != null;
+        AliasOrIndex aliasOrIndex = getAliasAndIndexLookup().get(aliasOrIndexName);
+        if (aliasOrIndex == null) {
+            throw new IndexNotFoundException(aliasOrIndexName);
+        }
+
+        // regular index
+        if (indices.containsKey(aliasOrIndexName)) {
```

Comment:
nit: should we prefer this as if it exists? I suspect it's the more common path?